src/SdfEntityCreator.cc

```diff
@@ -216,6 +220,41 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Model *_model)
 {
   IGN_PROFILE("SdfEntityCreator::CreateEntities(sdf::Model)");
 
+  // todo(anyone) Support multiple canonical links in nested models
+  // This version of CreateEntties keeps track whether or not to create a
+  // canonical link in a model tree using the second arg in this recursive
+  // function. We also override child nested models static property if parent
+  // model is static
+  auto ent = this->CreateEntities(_model, true, false);
+
+  // Load all model plugins afterwards, so we get scoped name for nested models.
+  for (const auto &[entity, element] : this->dataPtr->newModels)
```

Comment:
This was mainly done so the all the ancestor models are loaded first before loading the nested model's plugin so that it's able to get the scoped name. As for the order of loading the plugins, I think it maybe fine but not 100% certain. I just checked the behavior in gazebo classic and it's also loading the parent model's plugin first before the children's plugins. So we could give this a try and come back if we run into issues in the future?

Reply:
Sounds good to me.